crawlerr/services/crawler_service.py

```
@@ -245,92 +252,36 @@ async def crawl_website(
-                stats.total_pages_failed = len([r for r in crawl_results if not r.success])
-                
-                logger.info(f"Crawled {len(pages)} pages successfully, {stats.total_pages_failed} failed")
+            else:
+                # Use sitemap URLs - batch crawl approach
+                urls_to_crawl = list(discovered_urls)[:request.max_pages]
+                pages = await self._batch_crawl(urls_to_crawl, stats)
+            
+            # Set result pages and update statistics  
+            result.pages = pages
+            stats.total_pages_crawled = len(pages)
+            logger.info(f"Crawled {len(pages)} pages successfully, {stats.total_pages_failed} failed")
```

Ruff (0.12.2)

Logging statement uses f-string

(G004)